libgalois/include/katana/PropertyManager.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+#include "katana/Manager.h"
+#include "katana/MemoryManager.h"
+
+namespace katana {
+
+/// Manager for property memory
+class PropertyManager : public Manager {
+public:
+  /// Returns the coarse category of memory use
+  ///   e.g., property for the property manager
+  virtual const std::string& MemoryCategory() { return "property"; }
+
+  /// Free standby memory, attempting to free \p goal bytes.
+  /// Returns the number of bytes freed, which can only be less than goal if the
+  /// manager's standby total is less than goal.
+  virtual count_t FreeStandbyMemory(count_t goal);
+
+  void AddProperty(const katana::Uri& property_path) {
+    auto property = cache_->Get(property_path);
+    if (property.has_value()) {
+      auto bytes = katana::ApproxTableMemUse(property.value());
+      MM().StandbyToActive(this, bytes);
+      property_cache_->Evict(property_path);
+    }
+  }
+
+  void PropertyLoadedCallback(const std::shared_ptr<arrow::table>& property) {
+    MM.BorrowActive(*this, property.size());
+  }
+
+  void UnloadProperty(const std::shared_ptr<arrow::table>& property) {
+    auto granted = MemorySupervisor().ActiveToStandby(*this, property.size());
+    if (granted >= property.size()) {
+      property_cache_->Insert(property);
+    } else {
+      MM.ReturnActive(property.size());
+    }
+  }
+
+  count_t FreeStandbyMemory(count_t goal) {
+    count_t total = 0;
+    if (goal >= cache_.size()) {
+      total = cache_.size();
+      MM.ReturnStandby(total);
+      cache_.clear();
+    } else {
+      total = cache_->Reclaim(goal);
+      MM().ReturnStandby(total);
```

Comment:
If my point on callers of this function hold, then this call should be moved to the supervisor's calling code.

Reply:
I'm not sure what you mean.  The supervisor calls FreeStandbyMemory in a manager, and how much it asks for from each manager is one of its most important policy decisions.

Reply:
Meant to delete that comment. Please ignore it.